File: VocabluaryTrainer/Classes/Word.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VocabularyTrainer
{
    public class Word
    {
        public string value { get; set; }
        public string translation { get; set; }
        public string pronounciation { get; set; }
        public int priority { get; set; }
        public int tries { get; set; }
        public int right { get; set; }
    }
}

File: src/scripts/Word.js
let Word = {
    text : '',
    translation : '',
    pronounciation : '',
    right : 0,
    tests : 0,
    priority : 0,

    get percent() {
        let right = this.right || 0;
        let tests = (this.tests && this.tests > 0) ? this.tests : 1;

        return Math.round(right / tests * 100);
    }
}

export default Word;
